service/src/main/java/bio/terra/workspace/service/job/JobBuilder.java

```diff
@@ -70,6 +88,41 @@ public String submit() {
    */
   @Traced
   public <T> T submitAndWait(Class<T> resultClass) {
-    return jobServiceRef.submitAndWait(flightClass, jobParameterMap, resultClass, jobId);
+    validateAndDefault();
+    return jobService.submitAndWait(flightClass, jobParameterMap, resultClass, jobId);
+  }
+
+  // Check the inputs, supply defaults and finalize the input parameter map
+  private void validateAndDefault() {
```

Comment:
what do you think we rename this method to populateInputParams? there's not really any validation